spec/rails_helper.rb

```diff
@@ -123,3 +123,14 @@
 # The en-GB locale gives us UK counties (used by the LocalDeliveryUnit factory)
 # See all the things we gain here: https://github.com/faker-ruby/faker/blob/master/lib/locales/en-GB.yml
 Faker::Config.locale = 'en-GB'
+
+# Define a global sequence for generating NOMIS Offender IDs
+# NOMIS offender IDs follow the format: <letter><4 numbers><2 letters> (all uppercase)
+FactoryBot.define do
+  sequence :nomis_offender_id do |seq|
+    index = seq - 1 # because seq starts at 1, not 0
+    number = "%04d" % (index / 26) # zero-pad to 4 digits, e.g. 0001
+    letter = ('A'..'Z').to_a[index % 26]
+    "A#{number}A#{letter}"
```

Comment:
I deliberately used 'T' here tp try to indicate 'test' - IIRC A is used by real NOMIS (I think production) so maybe another letter might be prudent?